File: app/src/main/java/de/binder_net/tapper_test/FullscreenActivity.java
package de.binder_net.tapper_test;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Toast;

import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Locale;
import java.util.Random;
import java.util.Timer;
import java.util.TimerTask;

public class FullscreenActivity extends AppCompatActivity {


    public FullscreenActivity() throws IOException {
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_fullscreen);

        View mContentView = findViewById(R.id.fullscreen_content);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.hide();
        }

        mContentView.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                if (event.getAction() == MotionEvent.ACTION_DOWN){
//                    textView.setText("Touch coordinates : " +
//                            String.valueOf(event.getX()) + "x" + String.valueOf(event.getY()));

                    final Context context = getApplicationContext();
                    CharSequence text = String.format(Locale.ENGLISH, "X: %f | Y: %f", event.getX(), event.getY());
                    final int duration = Toast.LENGTH_SHORT;

                    Toast.makeText(context, text, duration).show();
                }

                return true;
            }
        });


        new Timer().scheduleAtFixedRate(new TimerTask() {
            @Override
            public void run() {

                long startTime = System.currentTimeMillis();


                try {
                    Process process = Runtime.getRuntime().exec("su");
                    DataOutputStream os = new DataOutputStream(process.getOutputStream());

                    Float x,y;

                    float minX = 0.0f;
                    float maxX = 1080.0f;
                    float minY = 0.0f;
                    float maxY = 1920.0f;

                    Random rand = new Random();

                    x = rand.nextFloat() * (maxX - minX) + minX;
                    y = rand.nextFloat() * (maxY - minY) + minY;

                    String cmd = String.format(Locale.ENGLISH, "/system/bin/input tap %f %f\n", x, y);
                    os.writeBytes(cmd);
                    os.writeBytes("exit\n");
                    os.flush();
                    os.close();
                    process.waitFor();

                    Log.d("My-Tag", cmd);
                } catch (IOException | InterruptedException e) {
                    e.printStackTrace();
                }

                long endTime   = System.currentTimeMillis();
                long totalTime = endTime - startTime;
                Log.d("My-Tag", String.format(Locale.ENGLISH, "Delta: %d", totalTime));
            }
        }, 1000, 5000);
    }


}

File: app/src/main/java/de/binder_net/tapper_test/FullscreenActivity.kt
package de.binder_net.tapper_test

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.MotionEvent
import android.widget.Toast
import java.io.DataOutputStream
import java.io.IOException
import java.util.*

class FullscreenActivity @Throws(IOException::class)
constructor() : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_fullscreen)

        val mContentView = findViewById(R.id.fullscreen_content)

        val actionBar = supportActionBar
        actionBar?.hide()

        mContentView.setOnTouchListener { _, event ->
            if (event.action == MotionEvent.ACTION_DOWN) {

                log("X: ${event.x} | Y: ${event.y}")
            }

            true
        }


        Timer().scheduleAtFixedRate(object : TimerTask() {
            override fun run() {

                val startTime = System.currentTimeMillis()

                try {
                    val process = Runtime.getRuntime().exec("su")
                    val os = DataOutputStream(process.outputStream)

                    val minX = 0.0f
                    val maxX = 1080.0f
                    val minY = 0.0f
                    val maxY = 1920.0f

                    val rand = Random()

                    val x = rand.nextFloat() * (maxX - minX) + minX
                    val y = rand.nextFloat() * (maxY - minY) + minY

                    val cmd = "/system/bin/input tap $x $y\n"
                    os.writeBytes(cmd)
                    os.writeBytes("exit\n")
                    os.flush()
                    os.close()
                    process.waitFor()

                    log(cmd)
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: InterruptedException) {
                    e.printStackTrace()
                }

                val endTime = System.currentTimeMillis()
                val totalTime = endTime - startTime
                log("Delta: $totalTime")
            }
        }, 1000, 5000)
    }

    fun log(message: String) {

        Log.d("FullscreenActivity", message)
    }
}
